compose.py

```diff
@@ -0,0 +1,248 @@
+    df = build.dockerfile_add_installation_linux(argmap, backends, endpoint)
+    with open(os.path.join(ddir, dockerfile_name), "a") as dfile:
+        dfile.write(df)
+
+
+### Create container with docker build
+def build_docker_image(ddir, dockerfile_name, container_name):
+    p = subprocess.Popen(['docker', 'build', '-t', container_name, '-f', \
+        os.path.join(ddir, dockerfile_name), '.'])
+    p.wait()
+    fail_if(p.returncode != 0, 'docker build {} failed'.format(container_name))
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    group_qv = parser.add_mutually_exclusive_group()
+    group_qv.add_argument('-q',
+                          '--quiet',
+                          action="store_true",
+                          required=False,
+                          help='Disable console output.')
+    group_qv.add_argument('-v',
+                          '--verbose',
+                          action="store_true",
+                          required=False,
+                          help='Enable verbose output.')
+    parser.add_argument('--output-name',
+                        type=str,
+                        required=False,
+                        help='Name for the generated Docker image. Default is "tritonserver".')
+    parser.add_argument(
+        '--work-dir',
+        type=str,
+        required=False,
+        help=
+        'Generated dockerfiles are placed here. Default to current directory.'
+    )
+    parser.add_argument(
+        '--container-version',
+        type=str,
+        required=True,
+        help=
+        'The version to use for the generated Docker image. If not specified the container version will be chosen automatically based on --version value.'
+    )
+    parser.add_argument(
+        '--enable-gpu',
+        action="store_true",
+        required=False,
+        help='Generate a Triton image that supports GPU'
+    )
+    parser.add_argument(
+        '--backend',
+        action='append',
+        required=False,
+        help=
+        'Include <backend-name> in the generated Docker image. The flag may be specified multiple times.'
+    )
+    parser.add_argument(
+        '--repoagent',
+        action='append',
+        required=False,
+        help='Include <repoagent-name> in the generated Docker image. The flag may be specified multiple times.')
+    parser.add_argument(
+        '--endpoint',
+        action='append',
+        required=False,
+        help='Include <endpoint-name> in the generated Docker image. The flag may be specified multiple times.')
+    FLAGS = parser.parse_args()
+    
+    if FLAGS.work_dir is None:
+        FLAGS.work_dir = "."
+    if FLAGS.output_name is None:
+        FLAGS.output_name = "tritonserver"
+
+    dockerfile_name = 'Dockerfile.compose'
+
+    if FLAGS.backend is None:
+        FLAGS.backend = []
+    if FLAGS.repoagent is None:
+        FLAGS.repoagent = []
+    if FLAGS.endpoint is None:
+        FLAGS.endpoint = []
+
```

Comment:
I was wrong before in saying we needed a --container-version flag. We don't want that flag. Instead we must get the container version from the branch that we are on. You will need to use the logic from build.py  that get the triton version from the TRITON_VERSION file, and then looks up in build.py:TRITON_VERSION_MAP to get the container version.